Library/NavigationTests.swift

```diff
@@ -38,6 +38,8 @@ public final class NavigationTests: XCTestCase {
     KSRAssertMatch(.project(.slug("project"), .root, refTag: nil),
                    "/projects/creator/project")
 
+    KSRAssertMatch(nil, "/projects/creator/project?token=4")
```

Comment:
Just proves that we get a `nil` match on this type of URL.

Reply:
Wonder whether we should also add `KSRAssertMatch(nil, "/projects/creator/project?ref=discovery&token=4")` for exhausted url example.